### ISSUE TITLE
Question about ChunksInRequestUploadHandler class

### ISSUE THREAD

**Issue by zedee** [pionl/laravel-chunk-upload#12]
Hi, I'd like to know why the "+1" on line 50 of `Handler/ChunksInRequestUploadHandler.php`:

```
// the chunk is indexed from zero (for 5 chunks: 0,1,2,3,4)
$this->currentChunk = $request->get("chunk") + 1;
```

in the constructor of the class.  Is it because compatibility for plupload and blueimp's uploaders? Because I'm experiencing problems with the chunk count by using `resumable.js` library. In that library, the first chunk already starts from `1` and not from `0`, getting a false `isLastChunk()` positive, rendering the file being merged before the last real chunk.

Any ideas?
Thanks,

**Comment by pionl** [comment]
HI @zedee, the plupload sends chunks indexes from zero. The `isLastChunk` checks, if the current chunk equals the `chunks` value. If there is 5 chunks, the last chunk is 4. That's why i change the chunk count from 1-5.

Does the `resumable.js` sends custom header in the request? So we can make a detection about the data.

**Comment by zedee** [comment]
Hi @pionl, yes. `resumable.js` sends custom headers in the request. This library allows to change the custom header key, and it's the first workaround I did to make it work.
Its default naming convention for the chunks (as of its [GitHub repo docs](https://github.com/23/resumable.js)) is:

- `chunkNumberParameterName The name of the chunk index (base-1) in the current upload POST parameter to use for the file chunk (Default: resumableChunkNumber)`.
- `totalChunksParameterName The name of the total number of chunks POST parameter to use for the file chunk (Default: resumableTotalChunks)`

**Comment by pionl** [comment]
Hi @zedee, thanks. I've added Handler for `resumable.js`. Use the default naming + update your dependency to master. You can use the dynamic handler or provide new `ResumableJSUploadHandler::class` handler. Could you let me know if it works?

**Comment by zedee** [comment]
Fantastic @pionl , what should I change to my `composer.json` to use that last version?

**Comment by zedee** [comment]
@pionl it worked fantastically! :1st_place_medal: 
Many, many thanks

**Comment by pionl** [comment]
@zedee Happy to hear :) Will release a new version with docs later on. Will let you know.
Best regards

**Comment by pionl** [comment]
Updated released `1.0.2`
